lib/utils.coffee

```diff
@@ -306,6 +309,12 @@ exports.requestAsync = (options) ->
 			headers: options.headers
 			uri: urlLib.parse(url)
 		return response
+	.catch (error) ->
```

Comment:
If you want to pull it out like that we can just implicitly return a nice ternary, rather than mutating `p`:

```coffeescript
...

if retriesRemaining > 0 then p.catch ->
    exports.requestAsync(options, retriesRemaining - 1)
else p
```

Thoughts? Otherwise I'd go for your 2nd option myself - the first for me is a little tricky to parse without having read much CS before. I'm still getting to grips with CS style though.